corehq/apps/ota/views.py

```diff
@@ -32,6 +42,70 @@ def restore(request, domain, app_id=None):
     return get_restore_response(domain, couch_user, app_id, **get_restore_params(request))
 
 
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def search(request, domain):
+    """
+    Accepts search criteria as GET params, e.g. "https://www.commcarehq.org/a/domain/phone/search/?a=b&c=d"
+    Returns results as a fixture with the same structure as a casedb instance.
+    """
+    criteria = dict(request.GET)
+    try:
+        case_type = criteria.pop('case_type')
+    except KeyError:
+        return HttpResponse('Search request must specify case type', status=400)
+
+    search_es = CaseSearchES()
+    search_es = search_es.domain(domain)
+    search_es = search_es.case_type(case_type)
+    config = CaseSearchConfig(domain=domain).config
+    fuzzies = config.get_fuzzy_properties_for_case_type(case_type)
+    for key, value in criteria.items():
+        search_es = search_es.case_property_query(key, value, fuzzy=(key in fuzzies))
+    results = search_es.values()
+    if should_use_sql_backend(domain):
+        cases = [get_instance_from_data(CommCareCaseSQLSerializer, result) for result in results]
+    else:
+        cases = [CommCareCase.wrap(flatten_result(result)) for result in results]
+    fixtures = CaseDBFixture(cases).fixture
+    return HttpResponse(fixtures, content_type="text/xml")
+
+
+@json_error
+@login_or_digest_or_basic_or_apikey()
+def claim(request, domain):
+    couch_user = CouchUser.from_django_user(request.user)
+    # TODO: use request.session to ensure only one extension case if multiple claims for the same beneficiary
+    #       come from the same AWW.
+    if request.method == 'POST':
+        host_id = request.POST['case_id']
```

Comment:
I could potentially see all of this going into a helper function `claim_case` that would live near where cases get closed, etc (I think in `cleanup.py` or something, although that name doesn't make sense any more if this would live there).